core/domain/classifier_services_test.py

```diff
@@ -92,6 +91,76 @@ def test_string_classifier_classification(self):
             self.assertTrue(
                 self._is_string_classifier_called('dunno, just guessed'))
 
+    def test_handle_creation_of_jobs_and_mappings(self):
+        """Test the handle_jobs_creation method and handle_mappings_creation
+        method by triggering update_exploration() method.
+        """
+        exploration = exp_services.get_exploration_by_id(self.exp_id)
+        state = exploration.states['Home']
+
+        # Now, there should be no jobs in the data store.
+        all_jobs = classifier_models.ClassifierTrainingJobModel.get_all()
+        self.assertEqual(all_jobs.count(), 0)
+
+        # Modify such that job creation is triggered.
+        state.interaction.answer_groups.insert(
+            3, state.interaction.answer_groups[1])
+        answer_groups = []
+        for answer_group in state.interaction.answer_groups:
+            answer_groups.append(answer_group.to_dict())
+        change_list = [{
+            'cmd': 'edit_state_property',
+            'state_name': 'Home',
+            'property_name': 'answer_groups',
+            'new_value': answer_groups
+        }]
+        with self.swap(feconf, 'ENABLE_STRING_CLASSIFIER', True):
+            exp_services.update_exploration(
```

Comment:
@seanlip 
EDIT to above comment: I resolved the issue. All the mapping creation and get_trainable_states_dict() method are called from the _save_exploration() function where we have access to the change_list - which is a list of the Exploration changes. So, I just passed change_list into the sub functions as another arg and was able to handle the State renaming case properly. This way, we avoid another GET operation as well.